src/include/migraphx/compile_options.hpp

```diff
@@ -32,9 +32,17 @@ inline namespace MIGRAPHX_INLINE_NS {
 
 struct compile_options
 {
-    bool offload_copy    = false;
+    /**
+     * Have MIGX allocate memory for parameters and add instructions
+     * to copy parameters and output to/from the GPU and host
+     */
+    bool offload_copy = false;
+
     bool fast_math       = true;
     bool exhaustive_tune = false;
+
+    /// Use the split_single_dyn_dim pass
+    bool split_single_dyn_dim = false;
```

Comment:
We can use this flag for now for testing, but ideally we should automatically split the dynamic dimension.

Reply:
I tried getting rid of the flag, but it's probably going to be needed for when we actually want things to run dynamically on the GPU. Removing it also breaks the `select_module` verify tests. Instead I just made it default to true for now.

Reply:
Going to change it back to defaulting to false. The `promote_literals` pass breaks too many tests otherwise.